### ISSUE TITLE
Provide proper label for featured image button

### ISSUE THREAD

**Issue by ocean90** [WordPress/gutenberg#10867]
**Describe the bug**
The featured image button doesn't have a proper label for screen readers. It currently announces the alt tag of the image which is "Featured Image" by default.

**To Reproduce**
1. Activate screen reader
2. Set a featured image
3. Focus the button/image

**Expected behavior**
A clear description of what the button does should be announced.

**Additional context**
https://core.trac.wordpress.org/changeset/36000
